core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/DebugUtil.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.commonutil;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Util methods for debugging
+ */
+public final class DebugUtil {
+    public static final String DEBUG = "gaffer.error-mode.debug";
+    public static final String DEBUG_DEFAULT = String.valueOf(false);
+    public static boolean isDebug;
+    private static final Logger LOGGER = LoggerFactory.getLogger(DebugUtil.class);
+
+    private DebugUtil() {
+        // this class should not be instantiated - it contains only util methods and constants.
+    }
+
+    public static boolean checkDebugMode() {
```

Comment:
I imagined:

```java
public static Boolean isDebug;

public static void isInDebugMode() {
    if(null == isDebug) {
        updateDebugMode();
    }

    return isDebug;
}
```

Every time toString is called on an object this method will be called - so that could be a lot. The value of the system property should never really change and if a user decides to change the system property value then it is easy enough for them to call updateDebugMode at the same time. You could have another method - setDebugMode() that sets the system property and updates the isDebug value.